### DISCUSSION TITLE
mdutil-gfm-table treats any table cell content as phrasing content

### DISCUSSION THREAD

**Discussion by ggoodman** [syntax-tree/unist#41] [Q&A] [answered]
I was attempting to programmatically build up a document using `mdast` that is comparable to an existing, hand-written markdown document. This hand-written document leverages some markdown features to allow embedding a subset of html within a table cell.

I noticed the following: https://github.com/syntax-tree/mdast-util-gfm-table/blob/704ed14ba30304e88b19250b2e3560665eeed3fa/to-markdown.js#L55-L60

Even if I attempt to embed a `html` node within the `tableCell`, the resulting printed document is stripped of the intended markup in the table cells.

**Comment by wooorm** [comment, marked as answer]
You cannot put flow content in GFM table cells, but phrasing is allowed:

```markdown
| > this is not a block quote |
| - |
| *this is phrasing* |
```

| > this is not a block quote |
| - |
| *this is phrasing* |

---

`html` is not necessarily flow, it could be phrasing too. Your last paragraph is thus surprising. Can you make that reproducible?

**Reply by ggoodman** [reply]
I thought a bunch about this overnight, and I think that I had unreasonable expectations, I will move my efforts over to the `hast` ecosystem. Let me explain:

We currently manage some hand-written markdown, structured as a markdown table with embedded html content in 'markdown' Contentful data type. Some layer in our system is able to correctly interpret and render this. This may seem like an unusual marriage and you would be right in thinking so; "why not just use html?" you might ask. There is some historical context to which I'm not privy but the good news is that there IS an 'html' data type that I could use instead.

What I think was unreasonable was asking for the `gfm`-specific plugin to produce markdown that is inconsistent with GitHub's spec. In other scenarios, it may well be useful to have a `mdast` extension that is able to nest block content in a table cell but that's another discussion.

Anyway, since we have an html data type, I'm going to shift my efforts over to `hast`. Thanks for clarifying!

**Reply by wooorm** [reply]
> In other scenarios, it may well be useful to have a mdast extension that is able to nest block content in a table cell but that's another discussion.

Agreed. I’d think of a different syntax (such as [this](https://github.com/zestedesavoir/zmarkdown/tree/HEAD/packages/remark-grid-tables#readme), but that project needs to be updated for micromark-in-remark). But then again: tables get complex quickly, so maybe an HTML table is a good alternative.

>Anyway, since we have an html data type, I'm going to shift my efforts over to hast. Thanks for clarifying!

Perfect! And good luck ;)
